ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -208,6 +212,63 @@ private[kvutils] case class ProcessTransactionSubmission(
           buildRejectionLogEntry(RejectionReason.Disputed("DuplicateKey: Contract Key not unique")))
     } yield r
 
+  /** Check that all parties mentioned in a transaction exist. */
+  private def validateParties: Commit[Unit] = {
+
+    def foldPartiesInTx[T](tx: GenTransaction.WithTxValue[NodeId, TContractId], z: T)(f: (T, String) => T): T =
+      foldPartiesInTxActors(tx, foldPartiesInTxVals(tx, z)(f))(f)
+
+    def foldPartiesInTxActors[T](tx: GenTransaction.WithTxValue[_, _], z: T)(f: (T, String) => T): T =
+      tx.fold(z) {
+        case (accum, (_, n)) => {
+          val parties = n match {
+            case c: NodeCreate[_, _] =>
+              (c.signatories union c.stakeholders)
+
+            case e: NodeExercises[_, _, _] =>
+              (e.actingParties union e.stakeholders)
+
+            case lk: Node.NodeLookupByKey[_, _] =>
+              lk.key.maintainers
+
+            case _ => Set[Ref.Party]()
+          }
+          parties.foldLeft(accum)(f)
+        }
+      }
+
+    def foldPartiesInTxVals[T](tx: GenTransaction.WithTxValue[_, TContractId], z: T)(f: (T, String) => T): T =
+      tx.foldValues(z) { (z, v) => foldPartiesInVal(v.value, z)(f) }
+
+    def foldPartiesInVal[T](v: Value[ContractId], z: T)(f: (T, String) => T): T =
+      v match {
+        case ValueParty(p) => f(z, p)
+        case other =>
+          InputsAndEffects.subValues(other).foldLeft(z) { (accum, v: Value[ContractId]) =>
+            foldPartiesInVal(v, accum)(f)
+          }
+      }
+
+    for {
+      allExist <- foldPartiesInTx(relTx, pure(true)) { (acc, p) =>
+          get(partyStateKey(p)).flatMap {
```

Comment:
Looks like `Commit` misses a `fold`. :slightly_smiling_face: Nothing to be solved here, just an observation.

Reply:
```suggestion
          get(partyStateKey(p)).flatMap(_.fold(pure(false))(_ => acc))
```
Would this do it?

Reply:
It's for sure more idiomatic and concise, although not super-easy on my eyes that are re-training to Scala :) but :+1: agree, I'll do it that way.